setup.py

```diff
@@ -20,7 +20,9 @@
 
 def _get_cpu_count():
     if platform.system() != "Windows":
-        return os.cpu_count()
+        cpu_count = os.cpu_count()
+        max_cores = min(cpu_count, int(os.getenv('MAX_BUILD_CORES', cpu_count)))
```

Comment:
I would wrap the cast to integer in a try [...] catch statement in case the user set it to something fishy.
```suggestion
        try:
            user_max_cores = int(os.getenv('MAX_BUILD_CORES'), cpu_count)
        except ValueError as e:
            # print useful error message
            raise e
        max_cores = min(cpu_count, user_max_cores)
```